mongodump/mongodump.go

```diff
@@ -511,17 +526,38 @@ func (dump *MongoDump) DumpIntent(intent *intents.Intent, buffer resettableOutpu
 	if err != nil {
 		return err
 	}
-	findQuery := &db.DeferredQuery{Coll: session.Database(intent.DB).Collection(intent.C)}
+	intendedDB := session.Database(intent.DB)
+	coll := intendedDB.Collection(intent.C)
+	// The storage engine cannot change from namespace to namespace,
+	// so we set it the first time we reach here, using a namespace we
+	// know must exist. If the storage engine is not mmapv1, we assume it
+	// is some modern storage engine that does not need to use an index
+	// scan for correctness.
+	// If ViewsAsCollections is set, all the collections will be views, and collStats will fail,
```

Comment:
I think you can still dump a mix of collections and views when`ViewsAsCollections` is set, right? So using the same strategy as mongoexport and checking `collectionInfo.IsView()` would be more accurate.

Reply:
You can't. I checked. If you do `ViewsAsCollections` it dumps *only* views :)

Reply:
Although, I think I'll do that, anyway!

Reply:
Ah interesting! I think we should make a ticket about that. Either to change the behavior or change the documentation. Because I certainly didn't expect that to be the behavior and the docs don't seem to clarify it.

Reply:
I made TOOLS-2450 to track this

Reply:
The `--help` message for it is clear, but the docs definitely are not!

Reply:
Looks like this comment is out of date now that we're not using `ViewsAsCollections`

Reply:
Oh yeah so it is. Yeah seems like a docs issue then.

Reply:
Updated the comment.